### ISSUE TITLE
MOM website update

### ISSUE THREAD

**Issue by aekiss** [mom-ocean/MOM5#222]
(shifted from https://github.com/OceansAus/access-om2/issues/94#issuecomment-387923632)

Just so we don't forget:

On Thu, Jun 22, 2017 at 10:12 PM, Stephen Griffies wrote:
Group,

I was going through the mom-ocean.science web site and have some suggestions for updates.

--It seems time to remove all mention of Breakaway Labs. It is mentioned at the top of the page as well as the link to the Model Development Lab.

--The COSIMA web site has a link to the mom-ocean.science web site, which is great. I suggest that mom-ocean.science should reciprocate. Perhaps there could be an "announcement" to point to the COSIMA project and its use/release of MOM5 based configurations. Should I presume you wish to have these configurations used by others outside of COSIMA?

Part of me wishes to get exposure to COSIMA among the MOM community. I also wish to let the MOM community know that there are still ongoing activities. The most recent update on mom-ocean.science was made 28 Nov, 2015 when we released the Southern Ocean visualization. We should let the MOM community know about ongoing work.

--We should provide a pointer from mom-ocean.science to the ENSO visualization from mom-ocean.science. This is another opportunity for an announcement on mom-ocean.science to let the community know there is stuff happening.

-- Please update the MOM history document with this one
https://drive.google.com/open?id=0ByuYBnZYpSJBMnVPUjZJS0xlMW8

**Comment by aidanheerdegen** [comment]
This needs @nicjhan to copy the website source off the host and make it a GitHub repo.

**Comment by aidanheerdegen** [comment]
@nicjhan provided me with the source for the current website, so I cloned it and made another heroku instance which is viewable  here:

https://mom5-ocn.herokuapp.com/web

The source has been put here:

https://github.com/mom-ocean/mom5-ocn

and any pushes to the master will trigger a site rebuild.

This is not ideal because heroku does not allow free hosting for teams, so I couldn't create a team that owned the app, which allow for shared administration etc. If someone wants to pony up the modest fees we could do that.

I also looked into creating a static version of the website so it could be hosted directly on github.io, but this was non-trivial, so I fell back to this solution.

Let us know what you think @StephenGriffies. We can try this for a while and see how it goes.

We can also add some CI checks so we don't bork the website with pushes to master.

**Comment by aidanheerdegen** [comment]
If you're happy @StephenGriffies, @nicjhan can change where mom-ocean.science is pointing to the new heroku instance.

**Comment by StephenGriffies** [comment]
Thanks for your efforts @aidanheerdegen and @nicjhan 

My biggest concern is that some of the present content on the website is obsolete and so needs to be updated.  Does the new approach allow us to edit the site?  

I am willing to pay a modest fee, depending on what "modest" means, and if doing so allows us to update the website as needed.

**Comment by aidanheerdegen** [comment]
The source for the site is now in the mom5-ocn repo. Any changes which are made and pushed to the master branch will be reflected in the heroku instance I linked above.

We need a README on the source repo to explain the basics of adding new content.

We can hack the clone I made and once you're happy with it, Nic change the mom-ocean.science domain name so that it resolves to the updated version.

Or if you prefer, make that change straight away and live hack the website. Your choice.

**Comment by aidanheerdegen** [comment]
So an example is the cosima-cookbook repo. The rendered docs are here:

http://cosima-cookbook.readthedocs.io/en/latest/

And the source is here:

https://github.com/OceansAus/cosima-cookbook/tree/master/doc

**Comment by StephenGriffies** [comment]
Under the mom5/doc repo, we should add all of the other mom.pdf fields presently on the website.  Those pdf files are important for many purpose, not the least of which is historical records.  

Other than those documents, I see little motivation to continue supporting the website.

**Comment by aidanheerdegen** [comment]
Actually we probably want an "organisation" website, which is even more straightforward, as it would live in it's own repo.

See here

https://pages.github.com

**Comment by StephenGriffies** [comment]
OK.  So let's move forward with something like the cookbook.  That seems more self-contained and under our control.  Can you set up a template for me to contribute to?  It is late here, but I hope to try working on this over the next days.

**Comment by aidanheerdegen** [comment]
Sure, I'll get something going

**Comment by StephenGriffies** [comment]
Or the organization website.  That seems good.  Thanks!

**Comment by aidanheerdegen** [comment]
Ok. There is a minimal website here:

https://mom-ocean.github.io

The source is here:

https://github.com/mom-ocean/mom-ocean.github.io

It is using the GitHub Pages built in support for Jekyll, which is a ruby based website templating engine thing.

It is using the minimal-mistakes theme, which has extensive documentation:

https://mmistakes.github.io/minimal-mistakes/docs/quick-start-guide/

Of particular interest, making posts:

https://mmistakes.github.io/minimal-mistakes/docs/posts/

Navigation

https://mmistakes.github.io/minimal-mistakes/docs/navigation/

and layouts

https://mmistakes.github.io/minimal-mistakes/docs/layouts/

I have added all the pdf documents from the old site, and the getting started doc, but not successfully linked that yet.

To make changes fork the repo, edit, commit and push back. It gets automatically rendered again. It is relatively straightforward to install the requisite ruby infrastructure to serve the site locally to debug and check it is correct. That shouldn't be necessary once it is all up and running, but it is not at that stage yet and I have already spent two days on this, so I will move on to more pressing matters in the short term.

Post problems to the issues page for the website repo and we can work on them collaboratively.

**Comment by aidanheerdegen** [comment]
I couldn't leave it so broken, so I've done a bit more work and it is sort of ok

https://mom-ocean.github.io

The individual post formatting is screwed up so I have made an issue

https://github.com/mom-ocean/mom-ocean.github.io/issues/1

I'll close this now.

**Comment by StephenGriffies** [comment]
Thanks again for your efforts @aidanheerdegen  I hope to have time to look this over in a few days...

**Comment by aidanheerdegen** [comment]
Hi @StephenGriffies,

The proto-website has all the manuals

https://mom-ocean.github.io/docs/

In some sense the website is complete, and at least can be updated. if you had enough time to just check it over and say you're happy with it as it is currently constituted, then @nicjhan can change where the mom-ocean.org address points to.

If you did want to make some more changes to the website you could then pay a student or similar to hack at it.

I'm not terribly wedded to it or anything, but there was a fair bit of time to get it to this stage, and I think it would be a shame to let it drop now.

Cheers

Aidan

**Comment by StephenGriffies** [comment]
Thanks @aidanheerdegen for this work.  I was under the impression that things were sitting on me to pull this together.  As it is presently organized, I think it is perfect.  

Now, please remind me where exactly this material will sit?  Will it be all on Github as the "readme" for the Github mom-ocean? 

One minor suggestion: we should have a link to the MOM6 Github repo, perhaps alongside the link to COSIMA.  

So in summary, I think this website should go live and we should abandon the original Breakaway Labs version of the mom-ocean.science website.  I prefer to keep all mom related material on Github or linked directly from the Github site.  

Am I making sense?

**Comment by aidanheerdegen** [comment]
Hi @StephenGriffies 

The website is it's own repository in the mom-ocean organisation

https://github.com/mom-ocean/mom-ocean.github.io

As far as this:

> Will it be all on Github as the "readme" for the Github mom-ocean?

The above website is an "organisation" website, so it is named for the whole Github organisation (mom-ocean), but there is no way that I can see to point to it from the top page of the mom-ocean GitHub organisation but I have created some pinned repositories to try and highlight them a bit when you hit the organisation page

When I said the website was mostly done, I didn't port over most of the blog posts, or the namelist stuff, which was done dynamically with the current (old) site.

> One minor suggestion: we should have a link to the MOM6 Github repo, perhaps alongside the link to COSIMA.

Feel free to change it. Those links are defined here:

https://github.com/mom-ocean/mom-ocean.github.io/blob/master/_data/navigation.yml

in the 'quicklinks' section.

Any opinions on this @nicjhan ?

**Comment by aidanheerdegen** [comment]
P.S. I updated the profile to include a link to the the mom-ocean.science website and updated the name, feel free to change it

![screen shot 2018-08-09 at 2 00 58 pm](https://user-images.githubusercontent.com/6063709/43877361-ab01dade-9bdc-11e8-8fc1-f2c85e556286.png)

(need a cool icon too)

**Comment by StephenGriffies** [comment]
I made some minor updates to the site, hopefully without messing anything.  Thanks for all your work.  

Is it now time to retire the site http://www.mom-ocean.science/web, pointing from that site to Github?

**Comment by aidanheerdegen** [comment]
Changes look great @StephenGriffies. Love the dark ocean pic. Very moody.

I'll close this issue and open a new one on the website repo to get the mom-ocean.science name pointing to the new website.

**Comment by StephenGriffies** [comment]
Thanks for the taking a look.  I too like the waves photo, though it is perhaps a bit too dark.  The website seems to darken it relative to the raw photo.  I may modify this photo in near future, unless you know of a way to render it a bit more brighter...
